### ISSUE TITLE
DefaultContentManager.CreateAsync doesn't publish any version after the first version

### ISSUE THREAD

**Issue by rjpowers10** [OrchardCMS/OrchardCore#9694]
### Describe the bug
In my code I have something like this
```
await _contentManager.UpdateValidateAndCreateAsync(contentItem, publish ? VersionOptions.Published : VersionOptions.Draft);
```

This works for the initial creation of the content item, but doesn't work if this is a new version for an existing content item.

### Expected behavior
If I pass VersionOptions.Published I expect that the new version will be published.

### Screenshots
I think the problem is in the `DefaultContentManager.CreateAsync` method.
![image](https://user-images.githubusercontent.com/68876423/122069936-8fcfa900-cdc3-11eb-9f95-203ee7e38cfd.png)

At line 431 the Published flag is set to true, but this only happens if the content item doesn't already have a content item version ID. Furthermore, at line 452 the Publishing and Published events are fired, but the Published flag on the content item is still false.

I'm creating my new version like this
```
await _contentManager.GetAsync(contentItemId, VersionOptions.DraftRequired);
```
 which appears to set the content item version ID right away.

### Workaround
Do something like the following
```
ContentValidateResult contentValidateResult = await _contentManager.UpdateValidateAndCreateAsync(contentItem, VersionOptions.Draft);
if (contentValidateResult.Succeeded && publish)
{
    await _contentManager.PublishAsync(contentItem);
}
```

**Comment by Skrypt** [comment]
I'm going to analyze this since me and @jtkech created this method. Maybe the issue is that you should not use it and call all the methods individually for your needs. I will try to analyze this and come up with a clear answer.

**Comment by Skrypt** [comment]
// Draft flag on create is required for explicitly-published content items
 contentItem.Published = !options.IsDraft;

I think this should fix it but I will try it first.

**Comment by Skrypt** [comment]
To create a new content item, I think you should use this instead.

```
var contentItem = await contentManager.NewAsync(contentType);
```

This because you want to trigger the appropriate event handlers (ActivatingAsync, ActivatedAsync, InitializingAsync, InitializedAsync). And also because it will create a new ContentItemVersionId every time. I think your issue comes from the fact that you are using `await _contentManager.GetAsync(contentItemId, VersionOptions.DraftRequired);` to create a new content item.

When we publish a content item we always create a draft then publish it. But you need to use the NewAsync() method to generate a new unique ContentItemVersionId. That's your issue there I'm pretty sure.

**Comment by Skrypt** [comment]
Actually just looked at the GetAsync() code and normally it should work fine too if you are doing : 

await _contentManager.GetAsync(contentItemId, VersionOptions.DraftRequired);

I need to understand what is the difference here then.

**Comment by Skrypt** [comment]
I agree that it should work because it will create a new ContentItemVersionId and call the VersioningAsync and VersionedAsync events. Need to see why it fails to publish then.

We need to do something like : 

```
if options.IsPublished then
    contentItem.Published = true;
end if
```

But need to make sure it won't break anything else. At this point if we pass VersionOptions.Published it could just be sanitary to make sure the ContentItem.Published is set to true just before triggering the events. Though, need to understand why we did not do this already.
